### DISCUSSION TITLE
Translating the logs

### DISCUSSION THREAD

**Discussion by Navid200** [NightscoutFoundation/xDrip#1964] [General]
The logs can help developers figure out what may be wrong.  The user can benefit from the logs as well.
There is a proposal to also translate the logs.  
Currently, only the settings titles and summaries are translated if you choose a specific language other than English.

Translating the logs will make them useless to someone who wants to help who happens to not speak the chosen language.
Not translating the logs could make the logs useless to someone who doesn't speak English.

Before we go too far into what would be good and what could be better, let me ask a question.

Can the logs technically be translated if we wanted to?
@jwoglom @tolot27 
Because if technically it cannot be done, then, there is no point in having this conversation and there is no decision to make.
So, would you please let me know what you think?

**Comment by tolot27** [comment]
> Can the logs technically be translated if we wanted to?

Yes, they can be translated and we may find a way not touching too many files (maybe just the `UserError` class) but this would result in a huge number of strings to translate requiring a lot of effort for the translators and approvers.
I believe logs should be for developers or experienced users only and mainly for support. I've never seen translated logs. Even Microsoft does not translate its logs.
If a user does not understand English it is simple to type the (typically short) text into an online translation tool.

**Comment by Navid200** [comment]
Alright, thanks.  I remember this came up in the past.  So, that's why I wanted this to be clear before we make any changes.

So, can we create a new class of logs that will only be used for a specific scenario where the developer intends the user, perhaps a child, to see an important note?
Say, only 0.1% of the logs would fall in this category.  
So, it would add to the job of the translators.  But, we wouldn't be translating all the existing neither all the new logs.

When a PR is opened, we can ask the developer to explain why a log has to be in this class before approving it.

**Reply by tolot27** [reply]
> So, can we create a new class of logs that will only be used for a specific scenario where the developer intends the user, ...

I recommend not to create a new class. Instead, we can create an additional function in the `UserError` class or just extend `UserErrorHigh` and `UserEventHigh`.
That is one use case, why it matters if `uel` or `ueh` is used (#1975).

**Comment by tolot27** [comment]
A bit off-topic but I've labeled this discussion accordingly. We could label any "important"/useful discussion like we did with the issues and PRs. This could help us later to identify enhancements, for instance.

**Comment by tzachi-dar** [comment]
I agree with the ideas in this discussion.
We should use uel and ueh to be translated as this is the only thing that a normal user will see. The nimber of messages there is small, so it is not a problem for translators.
We can also add color there.
